python/paddle/fluid/contrib/int8_inference/utility.py

```diff
@@ -423,107 +424,83 @@ def __get_quantize_dequantize_combination(self, program):
                     program.current_block().ops[value]._set_attr(
                         "force_fp32_output", True)
 
-            else:
-                if self._conv_op_index[index] + 1 < self._conv_op_index[index +
-                                                                        1]:
-                    for op_index in range(self._conv_op_index[index] + 1,
-                                          self._conv_op_index[index + 1]):
-                        op_type = program.current_block().ops[op_index].type
-                        if op_type in "conv2d" and self._conv_op_index[
-                                op_index] < self._conv_op_index[index + 1]:
-                            continue
-                        elif op_type in Calibrator.non_conv_int8_op_type:
-                            # dequantize_op_index.append(op_index) #disable dequantize if enable force_fp32_output
-                            program.current_block().ops[op_index - 1]._set_attr(
-                                "force_fp32_output", True)
-                            break
-                        else:
-                            program.current_block().ops[op_index - 1]._set_attr(
-                                "force_fp32_output", True)
+            elif self._conv_op_index[index] + 1 < self._conv_op_index[index +
+                                                                      1]:
+
+                program.current_block().ops[self._conv_op_index[
```

Comment:
Sorry, I missed that earlier. Why do we set the attribute here?

Reply:
It means there's a pattern that conv(int8)-->non-int8(op)--->...--->conv(int8) op. We don't insert the dequantize op but set the first conv(int8) op attribute force_fp32_output to True to get the fp32 output as the input of next non-int8 op.

Reply:
But the sequence
```
...conv2d->...->conv2d*->pool2d->conv2d->...
```
also falls into this case and the conv2d* op should not force fp32 output.